### ISSUE TITLE
Per-column labels are not supported

### ISSUE THREAD

**Issue by matias-sandell** [mherrmann/fullcalendar-columns#1]
Per-column labels would be very nice! Did you try to implement this and found it wasn't worth the trouble?

**Comment by mherrmann** [comment]
Hi Matias,

I agree it would be nice but I don't require them in my project. I also unfortunately won't have time in the coming weeks to implement this. It really shouldn't be too hard to implement though. So if you have the need I heartily encourage you to extend fullcalendar-columns and submit a pull request! :)

Best,
Michael

**Comment by matias-sandell** [comment]
OK, I'll have a go at it!

**Comment by JoeyHoutenbos** [comment]
Any update on this? The column label is required else it's unusable for me...

**Comment by matias-sandell** [comment]
Haven't had the time to start on this yet but please have a go at it if you like, @JoeyHoutenbos.

**Comment by aromka** [comment]
I added columns display in this PR - https://github.com/mherrmann/fullcalendar-columns/pull/4 - so if you're interested you can test it out. You will also notice a new variable `columnData` that will be reported during drag / drop, resize and time selection as well which is passed to columns.

Instead of setting `numColumns`, you would need to pass an array to `columns`, ie.:

```
columns: [ { id: 1, name: 'Col 1' }, { id: 2, name: 'Col 2' } ]
```

**Comment by JackIlPAZZO** [comment]
Hey #aromka, I'm interested in your solution. Could you please post a jsfiddle?

**Comment by mherrmann** [comment]
Thank you Roman! Two few questions:
- Why did you have to overwrite `reportSelection`? Is it something to do with column labels, or related to a different issue?
- Why did you have to set `result.endOrig = result.end.clone();` in `computeRange`?

Cheers! :)
M

**Comment by JackIlPAZZO** [comment]
How this code add the event to the calendar? 'Cause I don't see any method: http://fullcalendar.io/docs/event_data/addEventSource/
someone could help me to understand? I'm want using the aromka solution for have the resource label displayed.

**Comment by JackIlPAZZO** [comment]
Aromka I've contacted you on gmail. I want to propose my question:

http://stackoverflow.com/questions/33245038/how-to-redraw-fullcalendar-view

And I want also ask to you if I can get the column number when an eventi is moved to another column. In particolar if an event is on column 0 and the user nove the event on column 1 I want recognize this, how I can do this? Thank you.

**Comment by BluePanda1** [comment]
Hi aromka I want to ask you a question:
When I save an event on a column, for example column 0 it's possible check when a en event is moved on another column? If the event is moved on another column get the column number with resource label. That's only. I also want report you a bug, if I insert a lot of resources with a long name the label overlap the other cell, could you fix this. Thank you.

**Comment by Back2TheFuture** [comment]
Hi aromka I want to ask you a question:
When I save an event on a column, for example column 0 it's possible check when a en event is moved on another column? If the event is moved on another column get the column number with resource label. That's only. I also want report you a bug, if I insert a lot of resources with a long name the label overlap the other cell, could you fix this. Thank you.

PS: mrherrman I want improve your solution so don't delete my post 'cause other user could find usefull!!
Don't ban me to this repository again. How old are you? Show respect for anyone on your project and try to improve it.

**Comment by aromka** [comment]
@mherrmann 
- I overwrite `reportSelection` because I needed to get the `columnData` (which has the ID - in my project I had an option to toggle on / off columns, so I needed to know for which columns I should be rendering events).
- If you select time in a second column and look at the date - it gives you a fake date because of how this plugin renders the columns, so if you have 2 columns for `September 1` and select time - first column will return the proper date, but second column will return `September 2`, so `endOrig` should return you the proper date.

@Back2TheFuture - for long names you can just use css to prevent long names to overlap (ie. https://css-tricks.com/snippets/css/truncate-string-with-ellipsis/)
For moving the events to different columns - you should be able to get it from `select function` - something like: 

```
select: function(dtStart, dtEnd, jsEvent, event) {
    console.log(event.column, event.columnData);
}
```

**Comment by Back2TheFuture** [comment]
Hi @Aromka thanks for the support could you please provide a jsfiddle for the solution? Thank you :)

**Comment by mherrmann** [comment]
This has now been implemented in release v1.6. The API is a little different from @aromka's PR, for backwards-compatibility reasons. Also @aromka, I did not include all the changes you made because I am not sure yet whether they are "general" enough for everybody to be useful. I believe you should be able to get the same functionality with the new API by keeping a little more state in your "own" objects, rather than in `columnData` as you proposed. Thank you very much for your contribution. I'm not 100% sure the new API will fit your requirements as nicely, but I am certain that it will benefit others.
